### DISCUSSION TITLE
L4T Upgrades

### DISCUSSION THREAD

**Discussion by tpwrules** [OE4T/meta-tegra#823] [General]
I was interested in building Jetson images using a build system that's not Yocto, and was wondering how you guys handle upgrading and maintaining parity with L4T. Are all the Bitbake scripts manually written? I was hoping to not have to duplicate a bunch of the work if there is some source you use to generate everything.

**Comment by dwalkes** [comment]
AFAIK there's no real magic and everything is done manually.  You can look through some of the pull requests to see an example of what @madisongh does with new L4T releases.  Which build system are you thinking of targeting?

**Reply by madisongh** [reply]
At one point I did make a start on trying to script some of the steps, but never really got anywhere with it, due to differences in the way NVIDIA released things from version to version, on top of any content differences. Now that NVIDIA has added package feeds for the JetPack content, there's a bit more regularity that we could take advantage of - parsing the feeds for extracting package names and SHA256 hashes for automatically updating the recipes, for instance - but the L4T core would still require a lot of manual effort, and that's the bulk of the work, anyway.
